src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProvider.cs

```diff
@@ -22,13 +22,15 @@ public class SnapProvider : ISnapProvider
         private readonly IDbProvider _dbProvider;
         private readonly ILogManager _logManager;
         private readonly ILogger _logger;
+        private readonly HashSet<TrieNode> _boundaryProofNodes;
```

Comment:
Not sure whats going on here... A boundary proof node will no longer be boundary eventually. Don't think making it global a good idea, even if there is a mechanism to make it work. I'd prefer to keep it in the add account range.